### PR TITLE
Allow owner/admin to delete messages of a community

### LINKED ISSUE #14290: Allow owner/admin to delete messages of a community
As an admin/owner of a community
I should be able to delete a message of a community
so that I can make sure the content is relevant to users

### Description

An admin (or owner) should be given the option to delete messages from a community. 
This should be reflected on any member of the community.

If the role is `ROLE_ALL` or `ROLE_MODERATE_CONTENT` (this needs to be added), it should also be allowed to delete messages.

### PR DESCRIPTION
**Pull request by qfrank**
close #14290 

status: ready to review

notes: I've added a feature to help admin to make member as role of `moderator` so that it's very handy for testing, the feature need improvement later, e.g. admin can remove the role of member

### CODE REVIEW

File: src/status_im/ui2/screens/chat/messages/message.cljs [hunk 1]
```diff
@@ -51,11 +51,11 @@
 (def edited-at-text (str " ⌫ " (i18n/label :t/edited)))
 
 (defn message-status [{:keys [outgoing content outgoing-status pinned edited-at in-popover?]}]
-  (when-not in-popover? ;; We keep track if showing this message in a list in pin-limit-popover
+  (when-not in-popover?                                     ;; We keep track if showing this message in a list in pin-limit-popover
```
Comment (rasom) [inline src/status_im/ui2/screens/chat/messages/message.cljs]: It is likely that those whitespace changes are not necessary

Comment (qfrank) [reply src/status_im/ui2/screens/chat/messages/message.cljs]: yeah, i may need change the setting for my idea

**Review by OmarBasem — APPROVED** [review]
LGTM!

File: src/status_im/communities/core.cljs [hunk 2]
```diff
@@ -672,3 +675,23 @@
...
+
+(fx/defn add-role-to-member
+  {:events [:community.member/add-role]}
+  [_ community-id public-key role-id]
+  {::json-rpc/call [{:method     "wakuext_addRoleToMember"
+                     :params     [{:communityId community-id
+                                   :user        public-key
+                                   :role        role-id}]
+                     :on-success #()
+                     :on-error   #(log/error "failed to add role to member" community-id public-key role-id %)}]})
```
Comment (ilmotta) [inline src/status_im/communities/core.cljs]: This error log needs a bit of improvement. We can (and dare I say should) use data to give context in log messages. The `taoensso.timbre` library even encourages us to do that. Eventually, we should settle on some basic standards for logging messages, but for the time being, my suggestion is something along these lines:

```clojure
(log/error "Failed to add role to member"
           {:error        error
            :community-id community-id
            :public-key   public-key
            :role-id      role-id})
```

**Review by ilmotta — APPROVED** [review]
LGTM. I'm not commenting on the many things that could be refactored to follow the new guidelines because I think it's not the point of this PR, but that wouldn't hurt either ;)

**Review by ibrkhalil — APPROVED** [review]

**Comment by pavloburykh** [comment]
@qfrank Thanx for the PR. Please, take a look at the following issues:

### ISSUE 1 'Deleted for everyone' message remains undeleted after chat re-opening

Reproducible for all kinds of chats

**Steps:** 

1. Send message by User A to User B in any chat
2. Delete message for everyone by User A
3. Reopen chat by User A and User B and see if message remains deleted

**Actual result:** message is not deleted

**Expected result:** message remains deleted


https://user-images.githubusercontent.com/97245802/204802158-63cd4b23-f55a-4dfe-b1cf-66608821e3f6.mp4

[geth.log](https://github.com/status-im/status-mobile/files/10123395/geth.log)
[Status.log](https://github.com/status-im/status-mobile/files/10123397/Status.log)

**Comment by pavloburykh** [comment]
@qfrank also, could you please show, how to make member as role of moderator in current UI?

**Comment by qfrank** [comment]
> @qfrank also, could you please show, how to make member as role of moderator in current UI?


@pavloburykh

File: src/status_im/communities/core.cljs [hunk 3]
```diff
@@ -19,19 +19,21 @@
 
 (def crop-size 1000)
 
+(def member-role-moderator 3)
```
Comment (flexsurfer) [inline src/status_im/communities/core.cljs]: probably this should be in constants. where are 1 and 2?

**Comment by pavloburykh** [comment]
@qfrank ping me please when issues and review comments will be solved so I can re-test the PR.

File: src/status_im/communities/core.cljs (hunk 3 above)
Comment (qfrank) [reply src/status_im/communities/core.cljs]: Hey @flexsurfer , it's been a long time not c u your review :)  i thought 1 and 2 maybe defined in other PR in the future

Comment (flexsurfer) [reply src/status_im/communities/core.cljs]: yeah but still better to move it to constants

Comment (flexsurfer) [reply src/status_im/communities/core.cljs]: and also add community to the name

**Comment by pavloburykh** [comment]
> > @qfrank also, could you please show, how to make member as role of moderator in current UI?
> 
>  @pavloburykh

Thanx @qfrank Unfortunately, "Make moderator" feature does not work for me. Tapping "Make moderator" does not result in changing the role of the User, he is still unable to Delete for everyone messages in community, except of his own messages. 


https://user-images.githubusercontent.com/97245802/205032576-245031c9-cd3a-4c7f-ab9a-e75d8bb92063.mp4

[geth.log](https://github.com/status-im/status-mobile/files/10131579/geth.log)
[Status.log](https://github.com/status-im/status-mobile/files/10131580/Status.log)

**Comment by qfrank** [comment]
> > > @qfrank also, could you please show, how to make member as role of moderator in current UI?
> > 
> > 
> >  @pavloburykh
> 
> Thanx @qfrank Unfortunately, "Make moderator" feature does not work for me. Tapping "Make moderator" does not result in changing the role of the User, he is still unable to Delete for everyone messages in community, except of his own messages.
> 
>  making_member_moderator_bug.mp4 
> [geth.log](https://github.com/status-im/status-mobile/files/10131579/geth.log) [Status.log](https://github.com/status-im/status-mobile/files/10131580/Status.log)

let me reproduce it, okay, this happened after rebase, i'll look into it

**Comment by qfrank** [comment]
pls re-test :) @pavloburykh  thank you!

File: src/status_im/communities/core.cljs (hunk 3 above)
Comment (qfrank) [reply src/status_im/communities/core.cljs]: okay

**Comment by pavloburykh** [comment]
@qfrank thanx. Please, take a look at the new issue: 

### ISSUE 2 A regular member cannot delete his own message via "Delete for everyone option"

**Steps:**

1. Create community by User A
2. Create a new channel
3. Invite User B to a community 
4. Send message by User B in community channel
5. Try to "Delete for everyone" message by User B
6. See if message will be deleted

**Actual result:** User B cannot delete his own message via "Delete for everyone option"

**Expected result:** all the members, regardless of role that they have must have opportunity to delete for all their own messages



https://user-images.githubusercontent.com/97245802/205111918-ce9ae6de-e9cf-41a1-9137-b14ba46ef30c.mp4

**Comment by pavloburykh** [comment]
@qfrank thanx for your work. Ready for merge.

**Pull request merged**